geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/coprocessor/KryoLazyDensityCoprocessor.scala

```diff
@@ -67,47 +60,61 @@ class KryoLazyDensityCoprocessor extends KryoLazyDensityService with Coprocessor
   def getDensity(controller: RpcController,
                  request: KryoLazyDensityProto.DensityRequest,
                  done: RpcCallback[KryoLazyDensityProto.DensityResponse]): Unit = {
-    var scanner : InternalScanner = null
-    var filterList : FilterList = new FilterList()
 
-    val response: DensityResponse = try {
-      val options: Map[String, String] = deserializeOptions(request.getOptions.toByteArray)
-      val sft = SimpleFeatureTypes.createType("input", options(SFT_OPT))
-      var scanList : List[Scan] = List()
+    val options: Map[String, String] = deserializeOptions(request.getOptions.toByteArray)
+    val sft = SimpleFeatureTypes.createType("input", options(SFT_OPT))
 
-      if (options.containsKey(SCAN_OPT)) {
+    def aggregateResult(sf: SimpleFeature, result: DensityResult): Unit = writeGeom(sf, result)
 
+    val response: DensityResponse = try {
+
+      val scanList: List[Scan] = if (options.containsKey(SCAN_OPT)) {
         val decoded = org.apache.hadoop.hbase.util.Base64.decode(options(SCAN_OPT))
         val clientScan = ClientProtos.Scan.parseFrom(decoded)
-        val scan = ProtobufUtil.toScan(clientScan)
-        scanList ::= scan
-
+        List(ProtobufUtil.toScan(clientScan))
       } else {
-        scanList ::= new Scan()
+        List(new Scan())
       }
 
-      if (options.containsKey(FILTER_OPT)) {
-        filterList = FilterList.parseFrom(Base64.decode(options(FILTER_OPT)))
+      val filterList: FilterList = if (options.containsKey(FILTER_OPT)) {
+        FilterList.parseFrom(Base64.decode(options(FILTER_OPT)))
+      } else {
+        new FilterList()
       }
 
       val serializer = new KryoFeatureSerializer(sft, SerializationOptions.withoutId)
-      val densityResult: DensityResult = this.init(options, sft)
+      val densityResult: DensityResult = initialize(options, sft)
```

Comment:
think we need to instantiate a new KryoLazyDensityUtils to use here - and the main coprocessor class shouldn't extend it any more